cmd/startup/startup.go

```diff
@@ -0,0 +1,308 @@
+		return err
+	}
+	if err := s.writeKeyPair(&cs.Config.Certificates.ServiceSigningCa, path.Join(origin, "master"), "service-signer"); err != nil {
+		return err
+	}
+	if err := s.writeKeyPair(&cs.Config.Certificates.EtcdClient, path.Join(origin, "master"), "master.etcd-client"); err != nil {
+		return err
+	}
+	if err := s.writeKeyPair(&cs.Config.Certificates.AggregatorFrontProxy, path.Join(origin, "master"), "aggregator-front-proxy"); err != nil {
+		return err
+	}
+	if err := s.writeKeyPair(&cs.Config.Certificates.MasterKubeletClient, path.Join(origin, "master"), "master.kubelet-client"); err != nil {
+		return err
+	}
+	if err := s.writeKeyPair(&cs.Config.Certificates.MasterProxyClient, path.Join(origin, "master"), "master.proxy-client"); err != nil {
+		return err
+	}
+	if err := s.writeKeyPair(&cs.Config.Certificates.MasterServer, path.Join(origin, "master"), "master.server"); err != nil {
+		return err
+	}
+	if err := s.writePublicKey(&cs.Config.ServiceAccountKey.PublicKey, path.Join(origin, "master/serviceaccounts.public.key")); err != nil {
+		return err
+	}
+	if err := s.writeKey(cs.Config.ServiceAccountKey, path.Join(origin, "master/serviceaccounts.private.key")); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *startup) writeKeyPair(keypair *api.CertKeyPair, dir, name string) error {
+	if err := s.writeKey(keypair.Key, path.Join(dir, name+".key")); err != nil {
+		return err
+	}
+	if err := s.writeCert(keypair.Cert, path.Join(dir, name+".crt")); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *startup) writeCert(cert *x509.Certificate, destination string) error {
+	b, err := tls.CertAsBytes(cert)
+	if err != nil {
+		s.log.Errorf("writeCert %s %v", destination, err)
+		return err
+	}
+	s.log.Debugf("writing %s", destination)
+	return ioutil.WriteFile(filepath.Join(destination), b, 0600)
+}
+
+func (s *startup) writeKey(key *rsa.PrivateKey, destination string) error {
+	b, err := tls.PrivateKeyAsBytes(key)
+	if err != nil {
+		s.log.Errorf("writeKey %s %v", destination, err)
+		return err
+	}
+	s.log.Debugf("writing %s", destination)
+	return ioutil.WriteFile(filepath.Join(destination), b, 0600)
+}
+
+func (s *startup) writePublicKey(key *rsa.PublicKey, destination string) error {
+	b, err := tls.PublicKeyAsBytes(key)
+	if err != nil {
+		s.log.Errorf("writePublicKey %s %v", destination, err)
+		return err
+	}
+	s.log.Debugf("writing %s", destination)
+	return ioutil.WriteFile(filepath.Join(destination), b, 0600)
+}
+
+func (s *startup) init(ctx context.Context, log *logrus.Entry) error {
+	bsc, err := configblob.GetService("/etc/origin/cloudprovider")
+	if err != nil {
+		log.Fatal(errors.Wrap(err, "could not find storage account"))
+	}
+	s.blob = bsc.GetContainerReference(cluster.ConfigContainerName).GetBlobReference(cluster.ConfigBlobName)
+	s.log = log
+
+	return nil
+}
+
+func (s *startup) getBlob() (*api.OpenShiftManagedCluster, error) {
```

Comment:
-> library function to share with sync pod, don't duplicate

Reply:
i have moved this into util